File: languages/nim/test/testmips.nim
import sequtils, unittest, ../asmdot/mips

suite "test mips assembler":
  setup:
    var
      buf = newSeqOfCap[byte](100)

  test "should assemble single addi instruction":
    buf.addi(T1, T2, 0)

    check cast[seq[char]](buf) == toSeq("\x00\x00\x49\x21".items)

File: languages/ocaml/test/testmips.ml
open OUnit2

let suite = "mips suite" >::: [
  "should assemble single addi instruction" >:: (fun ctx ->
    let buf = Iobuf.create 4 in

    Mips.addi buf Reg.t1 Reg.t2 0;

    assert_equal ctx (Iobuf.to_string buf) "\x00\x00\x49\x21"
  );
];;

let () = run_test_tt_main suite ;;
